src/batch_request.cpp

```diff
@@ -17,14 +17,14 @@
 #include "batch_request.hpp"
 
 #include "execute_request.hpp"
+#include "external_types.hpp"
 #include "serialization.hpp"
 #include "statement.hpp"
-#include "types.hpp"
 
 
 extern "C" {
 
-CassBatch* cass_batch_new(CassBatchType type) {
+CassBatch* cass_batch_new(CassSession* session, CassBatchType type) {
```

Comment:
yes, that's part of it. I don't know what's going to need protocol version in the future and passing the protocol version alone doesn't seem like a good option. I was thinking this could be use to pass  around a custom allocator when that's implemented.

I'm open to better ways of handling this and sugggests. I just cant' think of a better way of handle this right now.